File: src/dataset/typescript/luhn_exercism-javascript-samajammin/gemini/GeminiFlash_luhn_v2.ts
export default class Luhn {
  public static valid(input: string): boolean {
    const stripped = input.replace(/\s/g, '');

    if (stripped.length <= 1 || !/^\d+$/.test(stripped)) {
      return false;
    }

    let sum = 0;
    for (let i = stripped.length - 1, j = 0; i >= 0; i--, j++) {
      let digit = parseInt(stripped[i], 10);
      if (j % 2 !== 0) {
        digit *= 2;
        if (digit > 9) {
          digit -= 9;
        }
      }
      sum += digit;
    }

    return sum % 10 === 0;
  }
}

File: src/dataset/c/luhn_Exercism_chriswilding/gemini/GeminiFlash_luhn_v2.c
#include "luhn.h"

#include <ctype.h>
#include <string.h>
#include <stdbool.h>

bool luhn(const char *number) {
    int sum = 0;
    int n = 0;
    int len = strlen(number);

    // Iterate backwards, skipping spaces and validating digits
    for (int i = len - 1; i >= 0; --i) {
        char c = number[i];

        if (c == ' ') {
            continue;
        }

        if (!isdigit(c)) {
            return false;
        }

        int digit = c - '0';

        // Apply Luhn algorithm
        if (n % 2 == 1) {
            digit *= 2;
            if (digit > 9) {
                digit -= 9;
            }
        }

        sum += digit;
        n++;
    }

    // Check if the number is valid
    return (n > 1 && (sum % 10 == 0));
}